logo.js

```diff
@@ -76,6 +76,26 @@ function LogoInterpreter(turtle, stream, savehook)
     return atom === match;
   }
 
+  function isPromise(value) {
+    return value && value.then && value.catch;
+  }
+
+  function promiseFinally(promise, finalBlock) {
```

Comment:
I'm not sure what should happen when finalBlock rejects... should everything reject (maybe covering up a previous rejection), or not...?  Semantically a bit vague to me.